File: Reststops.Core/Entities/Reststop.cs
using System.Collections.Generic;
using Reststops.Core.Enums;

namespace Reststops.Core.Entities
{
    public record Reststop
    {
        public ulong ID { get; init; }
        public string Name { get; init; }
        public double Latitude { get; init; }
        public double Longitude { get; init; }
        public ReststopType Type { get; init; }
        public Dictionary<string, string> Tags { get; init; }
    }
}

File: Reststops.Presentation.ReactNative/models/Reststop.ts
export interface Reststop {
    id:                      number;
    name:                    string;
    latitude:                number;
    longitude:               number;
    type:                    string;
    tags:                    {[key: string]: string};
    distanceInMeters:        number;
    detourDurationInSeconds: number;
}